pkg/kubeapiserver/admission/util/initializer.go

```diff
@@ -49,8 +50,45 @@ func IsUpdatingUninitializedObject(a admission.Attributes) (bool, error) {
 	if err != nil {
 		return false, err
 	}
-	if accessor.GetInitializers() == nil {
+	initializers := accessor.GetInitializers()
+	if initializers == nil || (len(initializers.Pending) == 0 && initializers.Result == nil) {
 		return false, nil
 	}
 	return true, nil
 }
+
+// IsFinalInitialization returns if the operation removes all pending initializers.
+func IsFinalInitialization(a admission.Attributes) (bool, error) {
+	if a.GetOperation() != admission.Update {
+		return false, nil
+	}
+	oldObj := a.GetOldObject()
+	oldAccessor, err := meta.Accessor(oldObj)
+	if err != nil {
+		return false, err
+	}
+	if oldAccessor.GetInitializers() == nil {
+		return false, nil
+	}
+	newObj := a.GetObject()
+	newAccessor, err := meta.Accessor(newObj)
+	if err != nil {
+		return false, err
+	}
+	initializers := newAccessor.GetInitializers()
+	if initializers == nil || (len(initializers.Pending) == 0 && initializers.Result == nil) {
+		return true, nil
+	}
+	return false, nil
+}
+
+// IsObjUninitialized returns if the new object still have pending initializers.
+func IsObjInitialized(a admission.Attributes) (bool, error) {
+	newObj := a.GetObject()
+	newAccessor, err := meta.Accessor(newObj)
```

Comment:
Perhaps this function should take an object, then it could be called on either old or new. And replace multiple lines of code repeating this logic earlier in this file.